lobster/schemas/gltf.fbs

```diff
@@ -0,0 +1,130 @@
+namespace gltf;
+
+table Asset {
+  generator:string;
+  version:string;
+}
+
+table Scene {
+  name:string;
+  nodes:[int];
+}
+
+table Node {
+  mesh:int;
+  name:string;
+  children:[int];
+  translation:[float];
+  rotation:[float];
+  scale:[float];
+}
+
+table Target {
+  node:int;
+  path:string;
+}
+
+table Channel {
+  sampler:int;
+  target:Target;
+}
+
+enum Interpolation : short { LINEAR }
+
+table Sampler {
+  input:int;
+  interpolation:Interpolation;
+  output:int;
+}
+
+table Animation {
+  channels:[Channel];
+  name:string;
+  samplers:[Sampler];
+}
+
+table ColorTexture {
+  index:int;
+  texCoord:int;
+}
+
+table MetallicRoughness {
+  baseColorTexture:ColorTexture;
+  metallicFactor:float;
+  roughnessFactor:float;
+}
+
+table Material {
+  name:string;
+  pbrMetallicRoughness:MetallicRoughness;
+}
+
+table Attributes {
+  POSITION:int;
+  NORMAL:int;
+  TANGENT:int;
+  TEXCOORD_0:int;
+}
+
+table Primitive {
+  attributes:Attributes;
+  indices:int;
+  material:int;
+  mode:int;
+}
+
+table Mesh {
+  name:string;
+  primitives:[Primitive];
+}
+
+table Texture {
+  source:int;
+}
+
+table Image {
+  mimeType:string;
+  name:string;
+  uri:string;
+}
+
+enum AccessorType : short { SCALAR, VEC2, VEC3, VEC4 }
+
+table Accessor {
+  bufferView:int;
+  byteOffset:int;
+  componentType:int;
+  type:string;
+  count:int;
+  max:[float];
+  min:[float];
+}
+
+table BufferView {
+  buffer:int;
+  byteLength:int;
+  byteOffset:int;
+  byteStride:int;
+}
+
+table Buffer {
+  byteLength:int;
+  uri:string;
+}
+
+table GLTF {
+  asset:Asset;
+  scene:int;
+  scenes:[Scene];
+  nodes:[Node];
+  animations:[Animation];
+  materials:[Material];
+  meshes:[Mesh];
+  textures:[Texture];
+  images:[Image];
+  accessors:[Accessor];
+  bufferViews:[BufferView];
+  buffers:[Buffer];
+}
+
+root_type GLTF;
```

Comment:
add a `file_extension gltfb` or whatever (glft binary?) here, that way `flatc` will save things with that extension rather than `bin`.. we are inventing a new file format here afterall.

Reply:
`fgltf`, `flatg`, `fbtf`, dunno. I haven't given the `flatc`-compiled binary version much thought. One exports their `gltf` and copies it into the folder, and that's that. Binary versions just obscure it a little. The bulk of the data is in the binary chunk anyways, which currently has to be external until I implement `glb`, so maybe call full binary `glfb` and compiled JSON + external binary something else?

Reply:
Not sure if there's a huge advantage to getting the external binaries internal.

Reply:
No performance advantage. It's a convenience format. On the other hand, `gltfb` is an inconvenience format.

Reply:
?

Reply:
Facebook uses the GLB format because it's a quick export from a user's 3D program of choice, potentially bundling images as well as geometry data into one file. When I was in high school I would have loved it. `gltfb` is slightly to the side of that, but I will implement it when I get a chance when I figure it out.